### PR TITLE
chore: add release-please automation

### PR DESCRIPTION
**Pull request by willarmiros**
## Which problem is this PR solving?

- Partially #605

## Short description of the changes

- Adds release-please automated independent release workflow so that individual packages can be released at their own cadence
- Pretty much a dupe of https://github.com/open-telemetry/opentelemetry-js/pull/2431 but for contrib

I tested these changes via an identical action in a dummy repo. You can see it completing a successful release here: https://github.com/willarmiros/test-package-lock-repo/runs/3618032476?check_suite_focus=true

### CODE REVIEW

File: .release-please-manifest.json [hunk 1]
```diff
@@ -0,0 +1,42 @@
+{
+  "detectors/node/opentelemetry-resource-detector-alibaba-cloud": "0.24.0",
+  "detectors/node/opentelemetry-resource-detector-aws": "0.24.0",
+  "detectors/node/opentelemetry-resource-detector-gcp": "0.24.0",
+  "detectors/node/opentelemetry-resource-detector-github": "0.24.0",
+  "metapackages/auto-instrumentations-node": "0.24.0",
+  "metapackages/auto-instrumentations-web": "0.24.0",
+  "packages/opentelemetry-browser-extension-autoinjection": "0.24.0",
```
Comment (willarmiros) [inline .release-please-manifest.json:8]: I'm not sure if this should be included, since it's `private`

File: release-please-config.json [hunk 2]
```diff
@@ -0,0 +1,47 @@
+{
+  "bootstrap-sha": "753e989de7b7663a389379fd1dfdaaba3d920c2e",
```
Comment (willarmiros) [inline release-please-config.json:2]: Currently this points here: https://github.com/open-telemetry/opentelemetry-js-contrib/commit/753e989de7b7663a389379fd1dfdaaba3d920c2e

But will need to be updated when there's a squash & merge.

File: .github/workflows/release-please.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+name: Run Release Please
+jobs:
+  release-please:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: google-github-actions/release-please-action@v2
+        with:
+          command: manifest
+          token: ${{secrets.RELEASE_PR_TOKEN}}
```
Comment (willarmiros) [inline .github/workflows/release-please.yml:14]: Will need the GH Token here @dyladan

**Comment by willarmiros** [comment]
I'll leave this as a draft since it's blocked on #657, which ought to be the final manual, repo-wide release in contrib. But it is ready for review, apart from updating `RELEASING.md`.

File: release-please-config.json (hunk 2 above)
Comment (dyladan) [reply release-please-config.json:2]: Just done

File: .github/workflows/release-please.yml (hunk 3 above)
Comment (dyladan) [reply .github/workflows/release-please.yml:14]: of course i'll add it

File: .release-please-manifest.json [hunk 4]
```diff
@@ -0,0 +1,42 @@
+{
+  "detectors/node/opentelemetry-resource-detector-alibaba-cloud": "0.25.0",
+  "detectors/node/opentelemetry-resource-detector-aws": "0.25.0",
+  "detectors/node/opentelemetry-resource-detector-gcp": "0.25.0",
+  "detectors/node/opentelemetry-resource-detector-github": "0.25.0",
+  "metapackages/auto-instrumentations-node": "0.25.0",
+  "metapackages/auto-instrumentations-web": "0.25.0",
+  "packages/opentelemetry-browser-extension-autoinjection": "0.25.0",
```
Comment (dyladan) [inline .release-please-manifest.json:8]: The manifest will be regenerated anyway on the next release so having extra entries is actually not a problem. Missing entries would be a bigger deal.

**Comment by willarmiros** [comment]
Do we want to also include releasing to NPM & GitHub in the release-please workflow as part of this PR, or should we do that in a separate PR? Reference: https://github.com/googleapis/release-please#automating-publication-to-npm

**Comment by dyladan** [comment]
@willarmiros whatever is easier for you

**Comment by willarmiros** [comment]
@dyladan I'll just update this PR to include the NPM and GH release steps

**Comment by willarmiros** [comment]
@obecny @vmarchaud I've finished the release-please automation for contrib and tested it as stated in description. PTAL if you have a chance :)

**Review by vmarchaud — APPROVED** [review]
lgtm

**Comment by obecny** [comment]
from code perspective it looks fine, but I will defer the review to @dyladan who is more familiar with releasing

File: .github/workflows/release-please.yml [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+      - uses: google-github-actions/release-please-action@v2
+        id: release
+        with:
+          command: manifest
+          token: ${{secrets.RELEASE_PR_TOKEN}}
+          default-branch: main
+
+      # The logic below handles the npm publication:
+      - name: Checkout Repository
+        if: ${{ steps.release.outputs.releases_created }}
```
Comment (dyladan) [inline .github/workflows/release-please.yml:20]: Is releases meant to be plural? I might be misremembering but i thought it was singular.

File: .github/workflows/release-please.yml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+          path: |
+            node_modules
+            package-lock.json
+            detectors/node/*/node_modules
+            metapackages/*/node_modules
+            packages/*/node_modules
+            plugins/node/*/node_modules
+            plugins/web/*/node_modules
+            propagators/*/node_modules
+          key: ${{ runner.os }}-${{ matrix.container }}-${{ hashFiles('**/package.json') }}
```
Comment (dyladan) [inline .github/workflows/release-please.yml:45]: I would add the name of the workflow/job to the key so it doesn't accidentally collide with other workflows.

File: .github/workflows/release-please.yml [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+          npx lerna bootstrap --no-ci
+
+      # Release Please has already incremented versions and published tags, so we just
+      # need to publish all unpublished versions to NPM here
+      # See: https://github.com/lerna/lerna/tree/main/commands/publish#bump-from-package
+      - name: Publish to NPM
+        if: ${{ steps.release.outputs.releases_created }}
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+        run: npx lerna publish from-package --no-push --no-git-tag-version --no-verify-access --yes
```
Comment (dyladan) [inline .github/workflows/release-please.yml:60]: Isn't there a --no-private flag as well?

**Review by dyladan — APPROVED** [review]
Looks good just some minor clarifications

File: .github/workflows/release-please.yml (hunk 5 above)
Comment (willarmiros) [reply .github/workflows/release-please.yml:20]: Yeah I believe their README is incorrect. I submitted a PR to correct it: https://github.com/googleapis/release-please/pull/1057

Based on my output from this test run it's plural: https://github.com/willarmiros/test-package-lock-repo/runs/3617992025?check_suite_focus=true#step:3:4

**Comment by willarmiros** [comment]
@dyladan is this ready for merging?

**Pull request merged**